TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/subsystems/WebcamAprilTags.java

```diff
@@ -0,0 +1,215 @@
+import org.firstinspires.ftc.vision.apriltag.AprilTagProcessor;
+import org.openftc.easyopencv.OpenCvInternalCamera;
+
+import java.util.List;
+
+public class WebcamAprilTags implements Subsystem {
+    private static WebcamAprilTags instance;
+    public enum Sections{
+        BLUE_ALLIANCE_SIDE_3_NEXUSES,
+        BLUE_MIDDLE_3_NEXUSES,
+        RED_ALLIANCE_SIDE_3_NEXUSES,
+        RED_MIDDLE_3_NEXUSES,
+        BLUE_BACKWARDS_NEXUSES,
+        RED_BACKWARDS_NEXUSES,
+        BLUE_PLATAFORM_NEXUSES,
+        RED_PLATAFORM_NEXUSES,
+        UNKNOWN_SECTION
+    }
+    private Telemetry telemetry;
+    private static AprilTagProcessor aprilTag;
+    static double CAM_DIST_TO_CENTER = 0;
+    // distancia da camera até o centro do robo
+    static double CAM_ANGLE_TO_GROUND =0;
+    //angulação da câmera (0º = paralelo ao chão, 90° = olhando para cima)
+    static double actualHead;
+    static double xtrans;
+    static double ytrans;
+    Pose2d prevDetectedPose;
+    private Sections currentSection = Sections.UNKNOWN_SECTION;
+    private static VisionPortal visionPortal;
+    public static AprilTagLibrary getFeedingTheFutureTagLibrary() {
+        return (new Builder())
+                .addTag(100, "Blue Nexus Goal - Field Center - Facing Platform", 160.0D, DistanceUnit.MM)
+                .addTag(101, "Red Nexus Goal - Field Center - Facing Platform", 160.0D, DistanceUnit.MM)
+                .addTag(102, "Red Nexus Goal - Field Center - Facing Food Warehouse", 160.0D, DistanceUnit.MM)
+                .addTag(103, "Blue Nexus Goal - Field Center - Facing Food Warehouse", 160.0D, DistanceUnit.MM)
+                .addTag(104, "Blue Nexus Goal - Field Edge - Alliance Station", 160.0D, DistanceUnit.MM)
+                .addTag(105, "Blue Nexus Goal - Field Edge - Center Field", 160.0D, DistanceUnit.MM)
+                .addTag(106, "Red Nexus Goal - Field Edge - Center Field", 160.0D, DistanceUnit.MM)
+                .addTag(107, "Red Nexus Goal - Field Edge - Alliance Station", 160.0D, DistanceUnit.MM)
+                .build();
+    }
+
+    /**
+     * Add telemetry about AprilTag detections.
+     */
+    @SuppressLint("DefaultLocale")
+    public Pose2d LocateWithAprilTag() {
+
+        List<AprilTagDetection> currentDetections = aprilTag.getDetections();
+        Sections section = Sections.UNKNOWN_SECTION;
+        currentSection = Sections.UNKNOWN_SECTION;
+
+        // Step through the list of detections and display info for each one.
+        if (currentDetections.size() > 0) {
+            double distX = 0;
+            double distY = 0;
+            double avgHead = 0;
+            for (AprilTagDetection detection : currentDetections) {
+                if (detection.metadata != null) {
+                    double xtag = 0;
+                    double ytag = 0;
+                    switch (detection.id) {
+                        //tem que ajustar os numeros para corresponder a arena
+                        case 100:
```

Comment:
what these values mean?
maybe add enum or constant where the name will explain it's functionality

Reply:
I ended up making them more readable as a result of another issue resolution, should be easier to understand now

Reply:
but they basically are the x and y coordinates of each april tag in the arena (which is mostly only used for telemetry and debugging)